src/Object/Api/Twitter/User.php

```diff
@@ -106,16 +119,16 @@ public function __construct(array $publicContact, array $apcontact = [], array $
 			unset($this->entities);
 		}
 		$this->description             = BBCode::toPlaintext($publicContact['about']);
-		$this->profile_image_url_https = $userContact['avatar'] ?? $publicContact['avatar'];
+		$this->profile_image_url_https = Contact::getAvatarUrlForUrl($publicContact['url'], $uid, Proxy::SIZE_MICRO);
```

Comment:
I'd prefer that you use the https://github.com/friendica/friendica/blob/d267ba999cf0b8691407e630563c251cb2ac68f9/src/Factory/Api/Twitter/User.php factory class and overwrite `publicContact['avatar']` at that class, so it's transparent for the current `Object` and we don't have any additional dependencies inside this `Object` class.

Reply:
I don't think that this would be a good way. `avatar` is a field in the contact table. We shouldn't change table fields because you won't expect this when working with a database record array.

Reply:
BTW: In that class we already are using `ContactSelector::networkToName`, `BBCode::toPlaintext` and `api_date`. So adding `Contact::getAvatarUrlForUrl` doesn't seem like a big deal to me.

Reply:
I thought that's because there's no alternative and should be replaced when possible as far as I can remind me discussing that point with @MrPetovan 

But I think we could think about it again when having real `Contact` Entities :-)